model/model.js

```diff
@@ -28,3 +29,22 @@ exports.selectArticles = () => {
       return rows;
     });
 };
+
+exports.selectArticleById = (article_id) => {
+  return checkIfValidId(article_id)
+    .then(() => {
+      return db.query(
+        `
+    SELECT * FROM articles
+    WHERE article_id = $1;
+    `,
+        [article_id]
+      );
+    })
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: "not found" });
```

Comment:
Just be careful that error messages don't become too general as this may make it hard to debug later on.